### PR TITLE
[huggingface_tensorflow, huggingface_pytorch] update for Transformers to 4.6.0 

### PR DESCRIPTION
**Pull request by philschmid**
… 4.6.0

*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

This PRs updates the `transformers` version inside the `huggingface` PyTorch and TensorFlow DLC.

*Tests run:*

https://github.com/huggingface/transformers/tree/master/tests/sagemaker

| ID                                  | Description                                                       | Platform                   | #GPUS | Collected & evaluated metrics            |
|-------------------------------------|-------------------------------------------------------------------|-----------------------------|-------|------------------------------------------|
| pytorch-transfromers-test-single    | test bert finetuning using BERT fromtransformerlib+PT             | SageMaker createTrainingJob | 1     | train_runtime, eval_accuracy & eval_loss |
| pytorch-transfromers-test-2-ddp     | test bert finetuning using BERT from transformer lib+ PT DPP      | SageMaker createTrainingJob | 16    | train_runtime, eval_accuracy & eval_loss |
| pytorch-transfromers-test-2-smd     | test bert finetuning using BERT from transformer lib+ PT SM DDP   | SageMaker createTrainingJob | 16    | train_runtime, eval_accuracy & eval_loss |
| pytorch-transfromers-test-1-smp     | test roberta finetuning using BERT from transformer lib+ PT SM MP | SageMaker createTrainingJob | 8     | train_runtime, eval_accuracy & eval_loss |
| tensorflow-transfromers-test-single | Test bert finetuning using BERT from transformer lib+TF           | SageMaker createTrainingJob | 1     | train_runtime, eval_accuracy & eval_loss |
| tensorflow-transfromers-test-2-smd  | test bert finetuning using BERT from transformer lib+ TF SM DDP   | SageMaker createTrainingJob | 16    | train_runtime, eval_accuracy & eval_loss |

*DLC image/dockerfile:*

The Hugging Face DLCs for PyTorch and TensorFlow

*Additional context:*
Bildschirmfoto 2021-04-06 um 14 20 34

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by saimidu** [comment]
The problem seems to originate from https://github.com/aws/deep-learning-containers/blob/master/test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smdp.py#L120-L130, where the folder `examples/question-answering` doesn't exist in the v4.6.0 tag on the `transformers` GitHub repo.

Could you please update the test there to use:

```
source_dir = (
    "./examples/question-answering"
    if Version(transformers_version) < Version("4.6")
    else "./examples/pytorch/question-answering"
)
...
estimator = HuggingFace(
    ...
    source_dir=source_dir,
    ...
)
```

**Comment by philschmid** [comment]
I updated the path! But it seems to fail somewhere else? Is there a way you can give me access?

File: test/sagemaker_tests/huggingface_pytorch/training/integration/local/test_single_machine_training.py [hunk 1]
```diff
@@ -59,9 +59,9 @@ def tokenize(batch):
     test_dataset = test_dataset.map(tokenize, batched=True, batch_size=len(test_dataset))
 
     # set format for pytorch
-    train_dataset.rename_column_("label", "labels")
+    train_dataset =  train_dataset.rename_column("label", "labels")
```
Comment (saimidu) [inline test/sagemaker_tests/huggingface_pytorch/training/integration/local/test_single_machine_training.py:62]: Is this change backward compatible between datasets versions 1.5 and 1.6?

File: test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smdp.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def test_smdp_question_answering(ecr_image, instance_type, py_version, sagemaker
     with timeout(minutes=DEFAULT_TIMEOUT):
         estimator = HuggingFace(
             entry_point='run_qa.py',
-            source_dir='./examples/question-answering',
+            source_dir='./examples/pytorch/question-answering',
```
Comment (saimidu) [inline test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smdp.py:93]: Could you please change this to be:
```
source_dir = (
    "./examples/question-answering"
    if Version(transformers_version) < Version("4.6")
    else "./examples/pytorch/question-answering"
)
...
estimator = HuggingFace(
    ...
    source_dir=source_dir,
    ...
)
```
so that we can continue to use these tests with HF Transformer versions 4.5 as well as 4.6?

File: test/sagemaker_tests/huggingface_pytorch/training/integration/local/test_single_machine_training.py (hunk 1 above)
Comment (philschmid) [reply test/sagemaker_tests/huggingface_pytorch/training/integration/local/test_single_machine_training.py:62]: Yes `train_dataset =  train_dataset.rename_column("label", "labels")` was introduced in `1.4.0`

**Review by saimidu — APPROVED** [review]
